src/bun.js/bindings/bindings.cpp

```diff
@@ -553,8 +566,10 @@ bool Bun__deepEquals(JSC__JSGlobalObject* globalObject, JSValue v1, JSValue v2,
             size_t count1 = 0;
 
             bool result = true;
-            if (o2Structure->inlineSize() + o2Structure->outOfLineSize() != o1Structure->inlineSize() + o1Structure->outOfLineSize()) {
-                return false;
+            if (strict) {
+                if (o2Structure->inlineSize() + o2Structure->outOfLineSize() != o1Structure->inlineSize() + o1Structure->outOfLineSize()) {
```

Comment:
properties don't have holes AFAIK

this will have a meaningful perf impact

Reply:
We can't exit early unless it's strict because the Jest implementation doesn't compare undefined properties for `toEqual()`
```js
expect({ a: undefined }).toEqual({ });
expect({ a: undefined }).not.toStrictEqual({ });
```

Reply:
oh i see